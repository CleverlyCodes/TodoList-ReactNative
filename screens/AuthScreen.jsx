import {
  Text,
  View,
  StyleSheet,
} from 'react-native';
import {
  Button,
} from 'react-native-elements';
import { theme } from '../constants/Theme';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: theme.white,
    alignItems: 'center',
    justifyContent: 'center',
  },
  appTitle: {
    fontSize: 20,
  },
  loginButtonStyle: {
    paddingHorizontal: 15,
    marginTop: 20,
  },
});

export default function Auth({onAuthenticate}) {
  return (
    <View style={styles.container}>
      <Text style={styles.appTitle}>
        HomeMaker - Grocery v0.1
      </Text>

      <Button
        title="Login"
        buttonStyle={styles.loginButtonStyle}
        onPress={onAuthenticate}
      />
    </View>
  );
}