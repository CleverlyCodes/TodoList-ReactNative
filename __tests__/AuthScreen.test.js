import React from "react";
import renderer from "react-test-renderer";
import AuthScreen from "../screens/AuthScreen.jsx"

jest.useFakeTimers();

describe("<AuthScreen />", () => {
  it('Renders action buttons to add list', async () => {
    const tree = renderer.create(<AuthScreen />).toJSON();

    await expect(tree).toMatchSnapshot();
  });
});