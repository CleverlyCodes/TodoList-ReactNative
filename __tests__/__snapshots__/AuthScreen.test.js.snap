// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AuthScreen /> Renders authentication screen 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 20,
      }
    }
  >
    HomeMaker - Grocery v0.1
  </Text>
  <View
    style={
      Array [
        Object {
          "borderRadius": 3,
          "overflow": "hidden",
        },
        Object {
          "borderRadius": 3,
        },
        undefined,
        false,
      ]
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "busy": false,
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#2089dc",
            "borderColor": "#2089dc",
            "borderRadius": 3,
            "borderWidth": 0,
            "flexDirection": "row",
            "justifyContent": "center",
            "marginTop": 20,
            "padding": 8,
            "paddingHorizontal": 15,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 18,
              "paddingVertical": 1,
              "textAlign": "center",
            }
          }
        >
          Login
        </Text>
      </View>
    </View>
  </View>
</View>
`;
